routes/recaptcha_solver.js

```diff
@@ -23,21 +23,19 @@ const DEFAULT_TIMEOUT = 1000;  // 1 second
  */
 
 async function action(page, request) {
-
     let recaptcha_data;
 
     if (request.body.solve_recaptcha) {
         recaptcha_data = await page.solveRecaptchas();
-    }
-    else {
+    } else {
         recaptcha_data = await page.findRecaptchas();
     }
 
     const waitOptions = request.body.waitOptions || { timeout: DEFAULT_TIMEOUT };
-    const close_page = request.query.closePage ||
+    request.query.closePage = request.query.closePage ||
```

Comment:
I'd like to consider request query immutable (even though it is not).
What do you think of closing the page here and adding page.isClosed() check in utils.performAction:
```javascript
        if (request.query.closePage && !page.isClosed()) {
            await page.close();
        }
        if (page.isClosed()) {
            delete response.pageId;
        }
```
That also affects closing pages in js actions.

Reply:
I'm OK with this.
This still has flexibility I wanted in the scrapy-puppeteer-service.
Tested changes with local spiders.